### ISSUE TITLE
Importing db_system fails because of db_home count change

### ISSUE THREAD

**Issue by lucianf** [terraform-providers/terraform-provider-oci#719]
I'm trying to import an existing DB System into an `oci_database_db_system` resource.  I've created the resource in Terraform and imported the existing db system OCID.

`terraform plan` informs that a new resource is required:

> db_home.#:                                                   "" => "1" (forces new resource)

However, there is (obviously!) already a `db_home` on the existing resource, which the provider doesn't seem to pick up (it sees `db_home.#` as 0) and as such forces creating a new resource, which is (obviously!) wrong.

This seems similar in nature to #700 so I wonder if there's a similar bug here.

### Terraform Version

<details>
Terraform v0.11.11
+ provider.oci v3.15.0
</details>

### OCI Provider Version

<details>
2019/02/26 11:57:48 [INFO] terraform-provider-oci 3.15.0
</details>

### Terraform Resource

<details>

```
resource "oci_database_db_system" "oci_xxx_db" {
  availability_domain = "${var.ad_list["AD1"]}"
  compartment_id      = "${oci_identity_compartment.xxx.id}"
  database_edition    = "ENTERPRISE_EDITION_HIGH_PERFORMANCE"

  db_home {
    database {
      admin_password = "XXX"
      db_name        = "XXX"
      db_workload    = "DSS"
      character_set  = "WE8ISO8859P15"
      ncharacter_set = "AL16UTF16"

      db_backup_config {
        auto_backup_enabled = true
      }
    }

    db_version   = "11.2.0.4"
    display_name = "XXX"
  }

  disk_redundancy = "HIGH"
  shape           = "VM.Standard2.4"
  subnet_id       = "${oci_core_subnet.xxx_sn.id}"

  ssh_public_keys = ["${chomp(file(var.authorized_keys_path))}"]
  display_name    = "XXX"

  hostname                = "XXX"
  data_storage_size_in_gb = "18432"
  license_model           = "LICENSE_INCLUDED"
  node_count              = "1"
  source                  = "NONE"
}
```

</details>

### Terraform Plan

<details>

```
-/+ oci_database_db_system.oci_xxx_db (new resource required)
      id:                                                          "ocid1.dbsystem.oc1.xxx.xxx" => <computed> (forces new resource)
      availability_domain:                                         "XXX" => "XXX"
      backup_subnet_id:                                            "" => <computed>
      cluster_name:                                                "" => <computed>
      compartment_id:                                              "ocid1.compartment.oc1..xxx" => "ocid1.compartment.oc1..xxx"
      cpu_core_count:                                              "4" => <computed>
      data_storage_percentage:                                     "80" => <computed>
      data_storage_size_in_gb:                                     "18432" => "18432"
      database_edition:                                            "ENTERPRISE_EDITION_HIGH_PERFORMANCE" => "ENTERPRISE_EDITION_HIGH_PERFORMANCE"
      db_home.#:                                                   "" => "1" (forces new resource)
      db_home.0.database.#:                                        "" => "1" (forces new resource)
      db_home.0.database.0.admin_password:                         <sensitive> => <sensitive> (forces new resource)
      db_home.0.database.0.backup_id:                              "" => <computed>
      db_home.0.database.0.backup_tde_password:                    <sensitive> => <computed> (attribute changed)
      db_home.0.database.0.character_set:                          "" => "WE8ISO8859P15" (forces new resource)
      db_home.0.database.0.db_backup_config.#:                     "" => "1" (forces new resource)
      db_home.0.database.0.db_backup_config.0.auto_backup_enabled: "" => "true" (forces new resource)
      db_home.0.database.0.db_name:                                "" => "UKMISADB" (forces new resource)
      db_home.0.database.0.db_workload:                            "" => "DSS" (forces new resource)
      db_home.0.database.0.ncharacter_set:                         "" => "AL16UTF16" (forces new resource)
      db_home.0.database.0.pdb_name:                               "" => <computed>
      db_home.0.db_version:                                        "" => "11.2.0.4" (forces new resource)
      db_home.0.display_name:                                      "" => "XXX" (forces new resource)
      disk_redundancy:                                             "HIGH" => "HIGH"
      display_name:                                                "XXX" => "XXX"
      domain:                                                      "xxx.xxx.oraclevcn.com" => <computed>
      freeform_tags.%:                                             "0" => <computed>
      hostname:                                                    "" => "XXX" (forces new resource)
      last_patch_history_entry_id:                                 "" => <computed>
      license_model:                                               "LICENSE_INCLUDED" => "LICENSE_INCLUDED"
      lifecycle_details:                                           "" => <computed>
      listener_port:                                               "1521" => <computed>
      node_count:                                                  "1" => "1"
      reco_storage_size_in_gb:                                     "3688" => <computed>
      scan_dns_record_id:                                          "ocid1.vcndnsrecord.oc1.xxx.xxx" => <computed>
      scan_ip_ids.#:                                               "0" => <computed>
      shape:                                                       "VM.Standard2.4" => "VM.Standard2.4"
      source:                                                      "" => "NONE" (forces new resource)
      sparse_diskgroup:                                            "" => <computed>
      ssh_public_keys.#:                                           "1" => "1"
      ssh_public_keys.0:                                           "ssh-rsa xxx" => "ssh-rsa xxx"
      state:                                                       "AVAILABLE" => <computed>
      subnet_id:                                                   "ocid1.subnet.oc1.xxx.xxx" => "ocid1.subnet.oc1.xxx.xxx"
      time_created:                                                "2019-xx-xx xx:xx:xx.xxx +0000 UTC" => <computed>
      version:                                                     "12.2.0.1.180717" => <computed>
      vip_ids.#:                                                   "0" => <computed>


Plan: 1 to add, 0 to change, 1 to destroy.
```

</details>

**Comment by lucianf** [comment]
Cc'ing @ravitandonrt just in case this is an easy one.

**Comment by alexng-canuck** [comment]
Hi @lucianf thanks for reporting this.

The db_system import behavior relies on the GetDbSystem APIs and a limitation of this API is that it doesn't surface information about associated DbHomes. We are investigating this.

In the meantime, if you wish to avoid the unexpected plan diffs, you can ignore changes to `db_home` for now by adding something like this to your db_system resource.

```
  lifecycle {
    ignore_changes = ["db_home"]
  }
```

**Comment by lucianf** [comment]
Could you provide a way to link to the `db_home` OCID, perhaps as a separate resource?  From there you can use the `GetDbHome` API to retrieve the details.

**Comment by lucianf** [comment]
Two more things:
- I couldn't get `lifecycle ignore_changes` to work in the resource - it seems to get ignored.
- Note that aside from `db_home`, the `hostname` attribute also forces a new resource.  It looks like the terraform provider is not pulling the existing `hostname` value (see my plan details above, the current hostname appears empty although in fact it isn't).  The REST API does return it correctly (confirmed with `oci-cli` and then `oci-curl` directly).

**Comment by lucianf** [comment]
Two more things:
- I couldn't get `lifecycle ignore_changes` to work in the resource - it seems to get ignored.
- Note that aside from `db_home`, the `hostname` attribute also forces a new resource.  It looks like the terraform provider is not pulling the existing `hostname` value.  The REST API returns it correctly (tried with `oci-cli` and then `oci-curl` directly).

**Comment by lucianf** [comment]
LE: This is strange - `ignore_changes` doesn't work for either `db_home` or `hostname` separately (the full diff is shown as if there was no ignore set), but it does work correctly when ignoring both together.

**Comment by alexng-canuck** [comment]
Hi @lucianf 

> Could you provide a way to link to the `db_home` OCID, perhaps as a separate resource? From there you can use the `GetDbHome` API to retrieve the details.

The `GetDbSystem` API doesn't give us a `db_home` OCID; so it's hard to determine which OCID to use to make the `GetDbHome` API call.

That being said, one possible solution is to provide both db system OCID and db home OCID at the time of `import`. That way, the provider can populate the db home sub-resource using `GetDbHome` API.

**Comment by mukkadinesh64** [comment]
hi,

I'm trying to create RAC with shape VM.Standard2.2 by default it will provision with 2 cores, but in state file it is showing as 4 core count and on consecutive apply it is failing 

* oci_database_db_system.this: Service error:InvalidParameter. The maximum CPU core count for VM.Standard2.2 DB System is 2.. http status code: 400. Opc request id: 5012f7ec5f80549d1801326422ad819b/CF8D3CB85EB58397A01421DAEE2D8EFD/386DDF49743513F16A6E84F920C9A94A

I tried putting property with 2 cores but still state file is showing 4. did anyone faced similar problem?

**Comment by alexng-canuck** [comment]
@lucianf This should now be resolved in the latest v3.22.0 OCI provider. See https://github.com/terraform-providers/terraform-provider-oci/blob/master/CHANGELOG.md#3220-april-10-2019 for more details.

You can read more about the import behavior from: https://www.terraform.io/docs/providers/oci/r/database_db_system.html#import

**Comment by alexng-canuck** [comment]
@mukkadinesh64 , can you open a new github issue for your question? That will make it easier for someone to find and respond to your question.
